src/Type/Php/JsonDecodeDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,86 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use Nette\Utils\Json;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\BooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ConstantTypeHelper;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\FloatType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use stdClass;
+
+final class JsonDecodeDynamicReturnTypeExtension implements DynamicFunctionReturnTypeExtension
+{
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		// @todo - finds only "assertType", but not "json_decode" :/
+		dump($functionReflection->getName());
+
+		return $functionReflection->getName() === 'json_decode';
+	}
```

Comment:
Here the `json_decode` is never passed :/

Reply:
To reproduce localy:

```bash
vendor/bin/phpunit tests/Type/Php/JsonDecodeDynamicReturnTypeExtensionTest.php
```

↓

![image](https://user-images.githubusercontent.com/924196/151775679-35ca9186-3d9a-4483-a0f1-c9cd98fcc0cf.png)

Reply:
Because there's already `JsonThrowOnErrorDynamicReturnTypeExtension` in phpstan-src whcih changes the result based on `JSON_THROW_ON_ERROR`. You need to modify it first.

Reply:
I'll try it